internal/controller/accessrequest_controller.go

```diff
@@ -189,51 +220,65 @@ func (r *AccessRequestReconciler) handlePermission(ctx context.Context, ar *api.
 		return "", fmt.Errorf("error verifying if subject is allowed: %w", err)
 	}
 	if !resp.Allowed {
-		err = r.updateStatus(ctx, ar, api.DeniedStatus, resp.Message)
+		rtHash := roleTemplateHash(rt)
+		err = r.updateStatus(ctx, ar, api.DeniedStatus, resp.Message, rtHash)
```

Comment:
I think we should only render the template if the user is allowed and we grant it access. The role template does not seem necessary when we are removing a user from the groups (only need role name) or refusing the request.

Reply:
Not sure I understand as there is no rendering at this point of the code. Rendering the template and updating the status with the new hash is necessary to make sure reconciliation requests are idempotent.